hasjob/views/listing.py

```diff
@@ -721,31 +715,33 @@ def confirm_email(domain, hashid, key):
     # and update post.datetime to utcnow() so it'll show on top of the stack
     # This function expects key to be email_verify_key, not edit_key like the others
     post = JobPost.query.filter_by(hashid=hashid).first_or_404()
-    if post.status in POSTSTATUS.GONE:
+    if post.state.GONE:
         abort(410)
-    elif post.status in POSTSTATUS.LISTED:
+    elif post.state.LISTED:
         flash("This job post has already been confirmed and published", "interactive")
         return redirect(post.url_for(), code=302)
-    elif post.status == POSTSTATUS.DRAFT:
+    elif post.state.DRAFT:
         # This should not happen. The user doesn't have this URL until they
         # pass the confirm form
         return redirect(post.url_for('confirm'), code=302)
-    elif post.status == POSTSTATUS.PENDING:
+    elif post.state.PENDING:
         if key != post.email_verify_key:
             return render_template('403.html.jinja2', description=u"This link has expired or is malformed. Check if you have received a newer email from us.")
         else:
             if app.config.get('THROTTLE_LIMIT', 0) > 0:
-                post_count = JobPost.query.filter(JobPost.email_domain == post.email_domain).filter(
-                    JobPost.status.in_(POSTSTATUS.POSTPENDING)).filter(
-                        JobPost.datetime > datetime.utcnow() - timedelta(days=1)).count()
+                post_count = JobPost.query.filter(
+                        JobPost.email_domain == post.email_domain
+                    ).filter(JobPost.state.POSTPENDING).filter(
+                        JobPost.datetime > datetime.utcnow() - timedelta(days=1)
+                    ).count()
                 if post_count > app.config['THROTTLE_LIMIT']:
                     flash(u"We have received too many posts with %s addresses in the last 24 hours. "
                         u"Posts are rate-limited per domain, so yours was not confirmed for now. "
                         u"Please try confirming again in a few hours."
                         % post.email_domain, category='info')
                     return redirect(url_for('index'))
             post.email_verified = True
```

Comment:
Move this line into the `post.confirm` transition method.

Reply:
In fact, all of this logic should move into the `post.confirm` method, but we can do that in the next PR where we remove the logic from views and move it into transitions.